cmd/main.go

```diff
@@ -2,115 +2,108 @@
-configuration repository.`,
-}
+	if err := config.BindPFlag("repo-dir", cmd.Flags().Lookup("repo-dir")); err != nil {
+		log.Fatalf("failed to bind flag repo-dir: %v", err)
+	}
 
-// Execute processes the opfcli command line.
-func Execute() {
-	if err := rootCmd.Execute(); err != nil {
-		log.Fatalf("Error: %s", err)
+	cfgFile, err := cmd.Flags().GetString("config-file")
+	if err != nil {
+		return err
 	}
-}
 
-func init() {
-	loglevel, err := strconv.Atoi(os.Getenv("OPF_LOGLEVEL"))
+	repoDirectory, err := cmd.Flags().GetString("repo-dir")
 	if err != nil {
-		loglevel = 1
+		return err
 	}
 
-	switch {
-	case loglevel >= 2:
-		log.SetLevel(log.DebugLevel)
-	case loglevel >= 1:
-		log.SetLevel(log.InfoLevel)
-	default:
-		log.SetLevel(log.WarnLevel)
+	if repoDirectory == "" {
+		repoDirectory, err = utils.FindRepoDir()
+		if err != nil {
+			repoDirectory, err = filepath.Abs(".")
+			if err != nil {
+				log.Fatalf("failed to determine repository directory: %v", err)
+			}
+		}
 	}
+	config.SetDefault("repo-dir", repoDirectory)
+	log.Debugf("using %s as repository directory", repoDirectory)
 
-	cobra.OnInitialize(initConfig)
+	if cfgFile != "" {
+		log.Debugf("using explicit configuration file %s", cfgFile)
+		config.SetConfigFile(cfgFile)
+	} else {
+		config.AddConfigPath(repoDirectory)
+	}
+	if err := config.ReadInConfig(); err != nil {
+		log.Debugf("failed to read configuration: %v", err)
+	}
+	log.Debugf("read configuration from %s", config.ConfigFileUsed())
 
-	rootCmd.PersistentFlags().StringVarP(
-		&cfgFile, "config", "f", "", "configuration file")
-	rootCmd.PersistentFlags().StringVarP(
-		&appName, "app-name", "a", "cluster-scope", "application name")
-	rootCmd.PersistentFlags().StringVarP(
-		&repoDirectory, "repodir", "R", "", "path to opf repository")
+	api.UpdateFromConfig(config)
 
-	if err = config.BindPFlag("app-name", rootCmd.PersistentFlags().Lookup("app-name")); err != nil {
-		panic(err)
-	}
+	return nil
 }
 
-func initConfig() {
+func NewConfig() *viper.Viper {
+	config := viper.New()
+
 	config.SetEnvPrefix("opf")
 	config.AutomaticEnv()
 	config.SetConfigName(".opfcli")
 
 	replacer := strings.NewReplacer("-", "_")
 	config.SetEnvKeyReplacer(replacer)
 
-	config.SetDefault("app-name", defaultAppName)
+	log.Debugf("returning new config")
 
-	if cfgFile != "" {
-		config.SetConfigFile(cfgFile)
-		if err := config.ReadInConfig(); err != nil {
-			log.Debugf("failed to read configuration: %v", err)
-		}
-	} else {
-		home, err := homedir.Dir()
-		if err == nil {
-			config.AddConfigPath(home)
-		}
+	return config
+}
 
-		if repoDirectory == "" {
-			repoDirectory, err = utils.FindRepoDir()
-			if err != nil {
-				repoDirectory, err = filepath.Abs(".")
-				if err != nil {
-					log.Fatalf("failed to determine repository directory: %v", err)
-				}
-			}
-		}
-		log.Debugf("using %s as repository directory", repoDirectory)
+func NewCmdRoot() *cobra.Command {
+	api := &api.API{}
```

Comment:
can you rename `api` here so it doesn't collide with the imported pkg?

Reply:
...and everywhere else, too, I guess.